pypdf/generic/_base.py

```diff
@@ -358,22 +365,26 @@ def clone(
         """clone object into pdf_dest"""
         return cast("FloatObject", self._reference_clone(FloatObject(self), pdf_dest))
 
-    def __repr__(self) -> str:
-        if self == self.to_integral():
-            # If this is an integer, format it with no decimal place.
-            return str(self.quantize(decimal.Decimal(1)))
+    def myrepr(self) -> str:
+        if float(self) == 0:
```

Comment:
Can we simply do the following?

```suggestion
        if self == 0:
```